### PR TITLE
income rpc methods

### PR DESCRIPTION
**Pull request by furkhat**

### CODE REVIEW

File: doc/ui/rpc.md [hunk 1]
```diff
@@ -97,6 +97,33 @@ This document describes a UI JSON RPC API located in "ui" namespace.
...
+*Method*: `getOfferingIncome`
+
+*Description*: Get total receipt balance from all channels of offering with given id.
+
+*Parameters*: 
+1. Password (string)
+2. Offering id (string)
+
+*Result*:
+1. Amount (number)
```
Comment (dzeckelev) [inline doc/ui/rpc.md]: *Result*: Amount (number)

File: ui/income.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import (
+	"database/sql"
+
+	"github.com/privatix/dappctrl/util/log"
+)
+
+// GetOfferingIncome returns total receipt balance from all channels of
+// offering with given id.
+func (h *Handler) GetOfferingIncome(
+	password, offeringID string) (uint, error) {
```
Comment (dzeckelev) [inline ui/income.go]: *uint

File: ui/income.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	logger := h.logger.Add("method", "GetOfferingIncome",
+		"offeringID", offeringID)
+
+	return h.income(logger, password,
+		`SELECT SUM(channels.receipt_balance)
+		   FROM channels
+		   WHERE channels.offering=$1`, offeringID)
```
Comment (dzeckelev) [inline ui/income.go]: line shifted

File: ui/income.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+// GetProductIncome returns total receipt balance from all channels of all
+// offerings with given product id.
+func (h *Handler) GetProductIncome(
+	password, productID string) (uint, error) {
```
Comment (dzeckelev) [inline ui/income.go]: *uint

File: ui/income.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+	return h.income(logger, password,
+		`SELECT SUM(channels.receipt_balance)
+		   FROM offerings
+		   JOIN channels ON channels.offering=offerings.id
+		     AND offerings.product=$1`, productID)
+}
+
+func (h *Handler) income(
+	logger log.Logger, password, query, arg string) (uint, error) {
```
Comment (dzeckelev) [inline ui/income.go]: *uint

File: doc/ui/rpc.md [hunk 6]
```diff
@@ -97,6 +97,33 @@ This document describes a UI JSON RPC API located in "ui" namespace.
...
+
+*Method*:   `getProductIncome`
+
+*Description*: Get total receipt balance from all channels of all offerings with given product id.
+
+*Parameters*: 
+1. Password (string)
+2. Product id (string)
+
+*Result*:   Amount (number)
```
Comment (ababo) [inline doc/ui/rpc.md]: Extra space

File: ui/income.go [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+		"productID", productID)
+
+	return h.income(logger, password,
+		`SELECT SUM(channels.receipt_balance)
+		   FROM offerings
+		   JOIN channels ON channels.offering=offerings.id
+		     AND offerings.product=$1`, productID)
+}
+
+func (h *Handler) income(
```
Comment (ababo) [inline ui/income.go]: Maybe "doSomethingWithIncome"?

**Pull request merged**